app/modules/map/opMapController.js

```diff
@@ -197,7 +197,12 @@ angular.module('opApp').controller('opMapController',
         $rootScope.$on('layer-selection-changed', function(event, layers) {
             var legends = [];
             angular.forEach(layers, function(layer) {
-                var legendGraphicUrl = opWebMapService.getLegendGraphicUrl(layer);
+                // layer is e.g. : dev:osg:archsites
+                //                 server:workspace:layername
+                var splitLayer = layer.split(':');
```

Comment:
Think about a way to write a utility function or two for handling server/typename extraction.